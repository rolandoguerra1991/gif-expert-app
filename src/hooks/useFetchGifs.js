import { useState, useEffect } from "react"
import getGifts from "../helpers/getGifs";

export const useFetchGifs = (category) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        setTimeout(() => {
            getGifts(category)
                .then(images => setState({
                    data: images,
                    loading: false,
                }));
        }, 3000);
    }, [category]);

    return state;
}