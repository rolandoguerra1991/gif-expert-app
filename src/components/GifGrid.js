import React, { Fragment } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem';

export default function GifGrid({category}) {

  const { data, loading } = useFetchGifs(category);

  const loadingHtML = <p>Loading</p>

  return (
    <Fragment>
      <h3>{category}</h3>
      { loading && loadingHtML }
      <div className='card-grid'>
        {
          data.map((img) => (
            <GifGridItem key={img.id} {...img} />
          ))
        }
      </div>
    </Fragment>
  )
}
