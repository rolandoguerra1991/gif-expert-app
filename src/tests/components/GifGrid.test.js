import GifGrid from "../../components/GifGrid";
import { shallow } from "enzyme";
import useFetchGifs from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en <GifGrid />", () => {
  test("debe de renderizar correctamente", () => {
    useFetchGifs.mockReturn({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={"Goku"} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
    const imgs = [{
      id: 'ABC',
      url: 'https://localhost/cualquier/cosa.jpg',
      title: 'Cualquier cosa'
    }]
    useFetchGifs.mockReturn({
      data: imgs,
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={"Goku"} />);
    expect(wrapper).toMatchSnapshot();
  });
});
