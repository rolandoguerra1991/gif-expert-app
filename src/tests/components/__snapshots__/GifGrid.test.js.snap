// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de renderizar correctamente 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
